src/builder/create_message.rs

```diff
@@ -61,6 +67,8 @@ pub struct CreateMessage {
     sticker_ids: Vec<StickerId>,
     #[serde(skip_serializing_if = "Option::is_none")]
     flags: Option<MessageFlags>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    attachments: Vec<MessageAttachment>,
```

Comment:
Also, it would be much clearer imo to rename this field to `attachments_metadata`, and rename `files` to `attachments`. That more closely reflects what they are: `attachments_metadata` is always empty, except very briefly before sending the request, when metadata is extracted out of `attachments` (new naming) into `attachments_metadata`, because that's where Discord expects the metadata to be.

Same proposal applies for all other builder structs that have both `attachments` and `files` currently